tools/check_format.py

```diff
@@ -86,15 +87,22 @@ def isBuildFile(file_path):
 
 
 def checkFileContents(file_path):
-  message = "%s has over-enthusiastic spaces:" % file_path
-  findSubstringAndPrintError('.  ', file_path, message)
+  findSubstringAndPrintError('.  ', file_path, "over-enthusiastic spaces")
+  for subdir in common.include_dir_order():
+    prefix = "#include <" + subdir
+    findSubstringAndPrintError(prefix, file_path, "envoy includes should not have angle brackets")
 
 
 def fixFileContents(file_path):
   for line in fileinput.input(file_path, inplace=True):
     # Strip double space after '.'  This may prove overenthusiastic and need to
     # be restricted to comments and metadata files but works for now.
-    sys.stdout.write("%s" % (line.replace('.  ', '. ')))
+    line = line.replace('.  ', '. ')
+    for subdir in common.include_dir_order():
```

Comment:
This seems like a lot of work to do for every line. We should guard it with a check that only lines with `#include <` prefix are under it.

Reply:
oh that's a good idea, and in fact there's a lot of performance improvement we can/should do.  Will update.